### DISCUSSION TITLE
Script-check

### DISCUSSION THREAD

**Discussion by ehsanijavad** [HewlettPackard/POSH-HPOneView#537] [Q&A] [answered]
Hi 
Frist of all thank you for your support.
I am trying to write a script, there are some challenges in my script. could you help me?

- I am creating a severprofile from a template, please let me know, how can I check when it will have finished with script. then I want to run the next command.
- I am using ks.cfg to install esxi, server will restart two times, do you have any idea to check it running or restarting. because after restarting I want to run some scripts.

Thanks in advance for your help and advices.

**Comment by ChrisLynchHPE** [comment, marked as answer]
If you want to post your script here, I can certainly help look at it.  Although, most New-OV Cmdlets will generate an Async task as noted in the Cmdlet documentation.  When it shows that one is returned, then use Wait-OVTaskComplete to monitor the task progress, whether it completes successfully or fails.  If it does fail, you will need to look within the `taskErrors` embedded property for the messages of why the task failed.

**Reply by ehsanijavad** [reply]
Thanks Chris, it works for me.
